core/IO.carp

```diff
@@ -21,7 +21,7 @@
   (doc read-file "returns the contents of a file passed as argument as a string.")
   (register read-file (Fn [&String] String))
   (doc exit "exit the current program with a return code.")
-  (register exit (Fn [Int] a))
+  (register exit (Fn [Int] ()) "exit")
```

Comment:
I think this was supposed to be `a` for a reason (to emulate "bottom"?)

Reply:
Yup, but it will emit invalid C by assigning a `void` function to a variable.

Reply:
Aha...

Reply:
Do you have a reproducible for when that happens?

Reply:
The simplest would probably be
```clojure
(let [x (IO.exit)] x)
; or
(and false (IO.exit))
```

Reply:
If it shouldn’t be used in these contexts, the easiest way to do that would probably to restrict its type to `()`.

Reply:
OK, I see. Yeah let's go with this then.